File: src/boj_1253.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class boj_1253 {
    public static void main(String[] args) {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        try {
            int N = Integer.parseInt(br.readLine());

            int[] arr = Arrays.stream(br.readLine().split(" "))
                    .mapToInt(Integer::parseInt)
                    .sorted()
                    .toArray();

            int answer = 0;

            for (int i = 0; i < N; i++) {

                int find = arr[i];
                int start = 0;
                int end = N - 1;


                while(start < end) {
                   if (find == arr[start] + arr[end]) {
                       if (start != i && end != i) {
                           answer++;
                           break;
                       } else if (start == i){
                           start++;
                       } else {
                           end--;
                       }
                   } else if (arr[start] + arr[end] < find) {
                       start++;
                   } else {
                       end--;
                   }
                }
            }
            System.out.println(answer);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: src/boj_1253.kt
fun main() {
    val br = System.`in`.bufferedReader();
    val n = br.readLine().toInt();
    val arr = br.readLine().split(" ").map { it.toInt() }.sorted()

    var answer = 0

    repeat(n) {
        var start = 0;
        var end = n - 1;

        while (start < end) {
            if (start == it) {
                start++;
                continue
            }
            if (end == it) {
                end--
                continue
            }

            val sum = arr[start] + arr[end]

            if(sum == arr[it]) {
                answer++
                break
            }

            if(sum < arr[it]) {
                start++
            } else {
                end--
            }
        }
    }
    println(answer)
}

